src/main/scala/beam/sim/BeamHelper.scala

```diff
@@ -667,10 +674,65 @@ trait BeamHelper extends LazyLogging {
     * @param outputDirectory output folder where full configs will be generated
     */
   private def writeFullConfigs(config: TypesafeConfig, outputDirectory: String): Unit = {
-    val configConciseWithoutJson = config.root().render(ConfigRenderOptions.concise().setFormatted(true).setJson(false))
+
+    val configResolveOptions = ConfigResolveOptions.defaults().setAllowUnresolved(true)
+
+    val templateConf = ConfigFactory
+      .parseResources("beam-template.conf")
+      .withoutPath("matsim.modules.vehicles.vehiclesFile")
+      .withoutPath("matsim.modules.transit.vehiclesFile")
+      .withoutPath("matsim.modules.counts.inputCountsFile")
+      .withoutPath("matsim.modules.strategy.planSelectorForRemoval")
+      .resolve(configResolveOptions)
+    val fullConfig = config.resolve().withFallback(templateConf).resolve()
+
+    val defaultConfig = fullConfig
+      .entrySet()
+      .asScala
+      .collect {
+        case entry if addKey(entry.getValue.unwrapped) =>
+          val unwrapped = entry.getValue.unwrapped()
+          val paramValue = unwrapped.toString
+          if (paramValue.contains("|")) {
+            entry.getKey -> actualValue(paramValue)
+          } else {
+            entry.getKey -> unwrapped
+          }
+      }
+      .toMap
+      .asJava
+    val defaultValues = ConfigFactory.parseMap(defaultConfig).resolve()
+    val configConciseWithoutJson =
+      defaultValues.root().render(ConfigRenderOptions.concise().setFormatted(true).setJson(false))
     writeStringToFile(configConciseWithoutJson, new File(outputDirectory, "fullBeamConfig.conf"))
+    writeStringToFile(defaultValues.root().render(), new File(outputDirectory, "fullBeamConfigJson.conf"))
+  }
 
-    writeStringToFile(config.root().render(), new File(outputDirectory, "fullBeamConfigJson.conf"))
+  private def actualValue(paramValue: String): Any = {
+    val value = paramValue.substring(paramValue.lastIndexOf('|') + 1).trim
+    if (paramValue.contains("int")) {
+      return value.toInt
+    }
+    if (paramValue.contains("double")) {
+      if ("Double.PositiveInfinity" == value) {
+        return Double.PositiveInfinity
+      }
+      if ("Double.NegativeInfinity" == value) {
+        return Double.NegativeInfinity
+      }
+      return value.toDouble
+    }
+    if (paramValue.contains("boolean")) {
+      return value.toBoolean
+    }
+    value
+  }
+
+  private def addKey(value: AnyRef): Boolean = {
```

Comment:
This threw me off...could you rename to something more like `shouldAddKey`?

Reply:
rename method name to shouldAddKey